src/transformers/models/xglm/modeling_xglm.py

```diff
@@ -338,9 +337,15 @@ def forward(
                     f"Attention mask should be of size {(bsz, 1, tgt_len, src_len)}, but is {attention_mask.size()}"
                 )
             attn_weights = attn_weights.view(bsz, self.num_heads, tgt_len, src_len) + attention_mask
+            attn_weights = torch.max(attn_weights, torch.tensor(torch.finfo(attn_weights.dtype).min))
             attn_weights = attn_weights.view(bsz * self.num_heads, tgt_len, src_len)
+            dtype_attn_weights = attn_weights.dtype
 
-        attn_weights = nn.functional.softmax(attn_weights, dim=-1)
+        # upcast to fp32 if the weights are in fp16. Please see https://github.com/huggingface/transformers/pull/17437
+        if dtype_attn_weights == torch.float16:
+            attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(dtype_attn_weights)
```

Comment:
Instead, we can change this part to 

```python
    if attn_weights.dtype == torch.float16:
        attn_weights = nn.functional.softmax(attn_weights, dim=-1, dtype=torch.float32).to(attn_weights.dtype)
```

Reply:
I think that we should fix the line on OPT too: https://github.com/huggingface/transformers/blob/34be08efcd4d318785b3eac592f27a3d5dd2144b/src/transformers/models/opt/modeling_opt.py#L221 in case `attention_mask` is set to `None` the forward pass will fail as described in https://github.com/huggingface/transformers/pull/18057#issuecomment-1224589697

Reply:
But I think that the issue has never been reported since `attention_mask` is never `None`: 
https://github.com/huggingface/transformers/blob/34be08efcd4d318785b3eac592f27a3d5dd2144b/src/transformers/models/opt/modeling_opt.py#L629

Reply:
Good catch! Surprisingly, we don't have test failure for OPT due to this.

Reply:
You answered my question before I asked it 😆